### PR TITLE
go.mod: bump Pebble to caecab07d786

### PR DESCRIPTION
**Pull request by nicktrav**
```
caecab07 db: export some types needed for using lazy values
4510a8c5 db: set sequence numbers in datadriven DB define commands
f07c5e5f db: use mergingIterLevel pointers as heap items
```

Release note: None.

Epic: None.

### CODE REVIEW

File: DEPS.bzl [hunk 1]
```diff
@@ -8990,6 +8990,16 @@ def go_deps():
             "https://storage.googleapis.com/cockroach-godeps/gomod/go.etcd.io/etcd/client/v2/io_etcd_go_etcd_client_v2-v2.305.0.zip",
         ],
     )
+    go_repository(
```
Comment (rickystewart) [inline DEPS.bzl:8993]: Looks like we now depend on both `go.etcd.io/etcd/raft/v3` as well as `go.etcd.io/raft/v3`. I don't know if this is a huge deal, but getting rid of the extra dependency would be good.

**Review by rickystewart — APPROVED** [review]
Keep an eye on Bazel CI *as well as* GitHub CI for this one please, I'm not sure if the new dependency will cause something to fail. You can pull me in with any questions.

**Comment by rickystewart** [comment]
Love to not have to deal with the `vendor` submodule any more 👍

File: DEPS.bzl (hunk 1 above)
Comment (nicktrav) [reply DEPS.bzl:8993]: Could this be some fallout from #93190? I assume the intention of that PR was to remove the dependency on the former? cc: @pavelkalinnikov

Let me see if I can resolve this issue separately.

**Review by sumeerbhola — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94096)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pavelkalinnikov and @rickystewart)

___
*[`DEPS.bzl` line 8993 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94096#-NJqEiAWZxbx3dKq-QoV-r1-8993:-NJqFpRP532YtGP2OGVk:b-wyj35z) ([raw file](https://github.com/cockroachdb/cockroach/blob/d3191c772e0cac0f1c917d399d2cd9f93debe453/DEPS.bzl#L8993)):*
<details><summary><i>Previously, nicktrav (Nick Travers) wrote…</i></summary><blockquote>

Could this be some fallout from #93190? I assume the intention of that PR was to remove the dependency on the former? cc: @pavelkalinnikov

Let me see if I can resolve this issue separately.
</blockquote></details>

I don't see any imports for the former on CockroachDB master (based on `git grep`), but my goland was quite insistently adding the import on a recent PR (so I had to close and reopen in emacs to undo). Is this a fallout of https://github.com/cockroachdb/cockroach/pull/93190?

**Review by sumeerbhola — APPROVED** [review]
@nicktrav thanks for this bump!
:lgtm: module the raft dependency issue

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94096#-:-NJqGl5M-jPcM_CgPNRP:byl7ija)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @pavelkalinnikov and @rickystewart)

**Comment by rickystewart** [comment]
My understanding is #93190 removed the dependency on the old package name `go.etcd.io/etcd/raft/v3` and replaced it everywhere with the new name `go.etcd.io/raft/v3`. I don't know where the import would have been resurrected from since `pebble`'s `go.mod` [doesn't seem to include it?](https://github.com/cockroachdb/pebble/blob/master/go.mod)

**Comment by rickystewart** [comment]
According to CI you need to `go mod tidy`.

**Comment by nicktrav** [comment]
> According to CI you need to go mod tidy.

Comes back clean for me when run locally :/ If I `go mod tidy` on `master`, it comes back with a diff. Same thing for me on both of my macOS and Linux machines.

I would not expect for _this PR_ to touch anything related to Raft. It should only be bumping the Pebble SHA. I ran `./scripts/bump-pebble.sh`.

**Comment by nicktrav** [comment]
I think I see what happened. There were generated proto files lying around that had the old import paths, which `go mod tidy` was seeing and pulling in the old dep. Running `./dev generate` fixed this.

I shall amend this commit.

**Review by rickystewart — APPROVED** [review]

**Comment by nicktrav** [comment]
The one test failure looks like #93958. Going to go ahead and land.

TFTRs!

bors r=rickystewart,sumeerbhola

**Pull request merged**